src/client/failure/event-adapter.ts

```diff
@@ -150,6 +158,15 @@ export function registerHooks(hookContext: HooksContext, events: Events): void {
         return { kind: 'retry' };
 
       case 'abort':
+        void events.emit('provider-switch', [
```

Comment:
Are we sure that provider switch is happening? Can be this abort from AbortPolicy?

Reply:
it's not happening. it's actually reporting that it can't switch, because there's not another provider configured. i suppose i should name the event better :thinking:

Reply:
Name confused me. On the other hand - if the only difference is the name 🤷